router/router.go

```diff
@@ -0,0 +1,182 @@
+package router
+
+import (
+	"encoding/json"
+	"fmt"
+	"iotdashboard/controller"
+	"log"
+	"net"
+	"net/http"
+)
+
+//TODO: maybe these ports should live in a config file?
+var httpPort = ":8080"
+var httpsPort = ":9090"
+
+var secHeaders = map[string]string{
+	"Strict-Tarnsport-Security": "max-age=63072000; includeSubDomains;",
+	"Content-Security-Policy":   "default-src 'self'",
+	"X-Frame-Options":           "DENY",
+	"X-Content-Type-Options":    "nosniff",
+	"Cache-Control":             "no-store",
+}
+
+// Create a struct to read the email and pass from the request
+type Credentials struct {
+	Email    string `json:"email"`
+	Password string `json:"password"`
+	CSRF     string `json:"csrf"`
+}
+
+func Start(certPath, keyPath string) {
+	fmt.Printf("Starting webserver ... \n")
+	//start listening for http to redirect to https
+	go http.ListenAndServe(httpPort, http.HandlerFunc(redirectTLS))
+	//start listening for https and handle requests
+	handleRequests(certPath, keyPath)
+}
+
+func handleRequests(certPath, keyPath string) {
+	mux := http.NewServeMux()
+	//TODO: move the finished react code into a local folder
+	mux.Handle("/", http.FileServer(http.Dir("../../../frontend-root/iot-dashboard/build/")))
+	mux.HandleFunc("/login", loginHandler)
+	mux.HandleFunc("/logout", logoutHandler)
+	mux.HandleFunc("/csrf", csrfHandler)
+
+	log.Fatal(http.ListenAndServeTLS(httpsPort, certPath, keyPath, mux))
```

Comment:
log.Fatal exits the program. Instead, return an error to the caller and let it handle it.